endpoint/domain_filter.go

```diff
@@ -78,9 +95,34 @@ func matchFilter(filters []string, domain string, emptyval bool) bool {
 	return false
 }
 
+// matchRegex determines if a domain matches the configured regular expressions in the DomainFilter.
+// The negativeRegex, if set, takes precedence over regex. Therefore,
+// matchRegex returns true when only regex regular expression matches the domain.
+// Otherwise, if either negativeRegex matches or regex does not match the domain, it will return false.
+func matchRegex(regex string, negativeRegex string, domain string) bool {
+	strippedDomain := strings.ToLower(strings.TrimSuffix(domain, "."))
+
+	if negativeRegex != "" {
+		match, err := regexp.MatchString(negativeRegex, strippedDomain)
+		if err != nil {
+			log.Errorf("Failed to filter domain %s with the regex-exclusion filter: %v", domain, err)
```

Comment:
We're masking the error here, what happens if there is something wrong? Does the program go ahead with a broken filter? We need to handle all errors.

Also, in Go normally when `err != nil` then you are not guaranteed that `match` is valid, it could as well be `nil`. 
Unit tests for this function will definitely help structuring the error handling better.

Reply:
I have switched the approach to compile the regex when parsing the arguments instead. So the regex will be validated and compiled on bootstrap, and then used here. If the regex is wrong, it panics and outputs whichever is the error with the regex at compile time